web/src/lib/components/shared-components/map/map.svelte

```diff
@@ -143,112 +142,143 @@
       country: featurePoint.properties.country,
     };
   };
+
+  $effect(() => {
+    map?.setStyle(styleUrl, {
+      transformStyle: (previousStyle, nextStyle) => {
+        if (previousStyle) {
+          // Preserves the custom map markers from the previous style when the theme is switched
+          // Required until https://github.com/dimfeld/svelte-maplibre/issues/146 is fixed
+          const customLayers = previousStyle.layers.filter((l) => l.type == 'fill' && l.source == 'geojson');
+          const layers = nextStyle.layers.concat(customLayers);
+          const sources = nextStyle.sources;
+
+          for (const [key, value] of Object.entries(previousStyle.sources || {})) {
+            if (key.startsWith('geojson')) {
+              sources[key] = value;
+            }
+          }
+
+          return {
+            ...nextStyle,
+            sources,
+            layers,
+          };
+        }
+        return nextStyle;
+      },
+    });
+  });
 </script>
 
-{#await style then style}
-  <MapLibre
-    {hash}
-    {style}
-    class="h-full"
-    {center}
-    {zoom}
-    attributionControl={false}
-    diffStyleUpdates={true}
-    on:load={(event) => event.detail.setMaxZoom(18)}
-    on:load={(event) => event.detail.on('click', handleMapClick)}
-    bind:map
-  >
-    {#snippet children({ map }: { map: maplibregl.Map })}
-      <NavigationControl position="top-left" showCompass={!simplified} />
-
-      {#if !simplified}
-        <GeolocateControl position="top-left" />
-        <FullscreenControl position="top-left" />
-        <ScaleControl />
-        <AttributionControl compact={false} />
-      {/if}
-
-      {#if showSettingsModal !== undefined}
-        <Control>
-          <ControlGroup>
-            <ControlButton on:click={() => (showSettingsModal = true)}><Icon path={mdiCog} size="100%" /></ControlButton
-            >
-          </ControlGroup>
-        </Control>
-      {/if}
-
-      {#if onOpenInMapView}
-        <Control position="top-right">
-          <ControlGroup>
-            <ControlButton on:click={() => onOpenInMapView()}>
-              <Icon title={$t('open_in_map_view')} path={mdiMap} size="100%" />
-            </ControlButton>
-          </ControlGroup>
-        </Control>
-      {/if}
-
-      <GeoJSON
-        data={{
-          type: 'FeatureCollection',
-          features: mapMarkers.map((marker) => asFeature(marker)),
+<!--  We handle style loading ourselves so we set style blank here -->
+<MapLibre
+  {hash}
+  style=""
```

Comment:
Delete this @zackpollard

Reply:
You can't, it's required in the type, there's even a comment to explain it :P